File: apps/ios/Barns/BarnsApp/Domain/UseCases/Auth/GetCurrentUserUseCase.swift
/// Returns the currently signed-in user, if any.
struct GetCurrentUserUseCase {
    private let authRepository: AuthRepository

    init(authRepository: AuthRepository) {
        self.authRepository = authRepository
    }

    func execute() async throws -> User? {
        try await authRepository.currentUser()
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/auth/GetCurrentUserUseCase.kt
package com.barns.app.domain.usecase.auth

import com.barns.app.domain.model.User
import com.barns.app.domain.repository.AuthRepository

/**
 * Returns the currently signed-in user, if any.
 */
class GetCurrentUserUseCase(
    private val authRepository: AuthRepository,
) {
    suspend fun execute(): User? = authRepository.currentUser()
}
